File: backend/src_api/schemas/onboarding.py
from pydantic import BaseModel, EmailStr, HttpUrl, Field
from src_api.schemas.auth import AccountBase, AccountRole
from enum import Enum
from typing import List, Optional



#database enums

class FundingStageEnum(str, Enum):
    PRE_SEED = "PRE_SEED"
    SEED = "SEED"
    PRE_SERIES_A = "PRE_SERIES_A"

class TRLEnum(str, Enum):
    IDEA = "IDEA"
    PROTOTYPE = "PROTOTYPE"
    PILOT = "PILOT"
    LIVE_PRODUCT = "LIVE_PRODUCT"
    SCALING = "SCALING"

class InvestorTypeEnum(str, Enum):
    ANGEL = "ANGEL"
    VC_FUND = "VC_FUND"
    FAMILY_OFFICE = "FAMILY_OFFICE"

#actual request/response schemas

class FounderOnboardingRequest(BaseModel):
    role: AccountRole = AccountRole.FOUNDER

    startup_name: str
    one_line_desc: str
    full_desc: str
    stage: FundingStageEnum
    trl: TRLEnum
    target_raise_inr: float
    min_cheque_inr: float

class InvestorOnboardingRequest(BaseModel):
    role: AccountRole = AccountRole.INVESTOR

    investor_type: InvestorTypeEnum
    brief_bio: str
    preferred_stages: List[FundingStageEnum]
    min_trl_accepted: TRLEnum

    min_cheque_inr: float
    max_cheque_inr: float

File: frontend/utils/api/onboarding.ts
import { apiClient } from "./client";
import { FundingStage, TRL, InvestorType } from "@/constants/enums";

export interface FounderOnboardingRequest {
  startup_name: string;
  one_line_desc: string;
  full_desc: string;
  stage: FundingStage;
  trl: TRL;
  target_raise_inr: number;
  min_cheque_inr: number;
}

export interface InvestorOnboardingRequest {
  investor_type: InvestorType;
  brief_bio: string;
  preferred_stages: FundingStage[];
  min_trl_accepted: TRL;
  min_cheque_inr: number;
  max_cheque_inr: number;
}

interface OnboardingResponse {
  status: number;
  message: string;
}

export const submitFounderOnboarding = (payload: FounderOnboardingRequest) =>
  apiClient.postJson<OnboardingResponse>("/onboarding/founder", payload);

export const submitInvestorOnboarding = (payload: InvestorOnboardingRequest) =>
  apiClient.postJson<OnboardingResponse>("/onboarding/investor", payload);
